### PR TITLE
Control exactly what we use in kubeadm

### PR DESCRIPTION
**Pull request by dims**
**What this PR does / why we need it**:

we recently got rid of extra arg validation for preflight check:
9f21f5dd1e3b9a42f5983c2d756c18f3a1f53b7a

the import-boss configuration will help us control exactly what
gets added/used in kubeadm. In this PR, we add a white-list of
the exact packages we use currently.

We will need to review these, make new PR(s) to  eliminate things
we don't want to be depending on (just like the cloud provider related code).

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: cmd/kubeadm/.import-restrictions [hunk 1]
```diff
@@ -0,0 +1,260 @@
+{
+	"Rules": [{
+		"SelectorRegexp": "k8s[.]io",
+		"AllowedPrefixes": [
+			"k8s.io/api/admission/v1beta1",
```
Comment (timothysc) [inline cmd/kubeadm/.import-restrictions]: Can't the prefixes be pruned down to the lowest common denominator for a bunch of the client stuff.  

e.g. k8s.io/api, k8s.io/apimachinery ...

Comment (timothysc) [reply cmd/kubeadm/.import-restrictions]: /approve

Comment (dims) [reply cmd/kubeadm/.import-restrictions]: yep, will tweak over the weekend.

File: cmd/kubeadm/.import-restrictions [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+				"k8s.io/kubernetes/pkg/apis/policy",
+				"k8s.io/kubernetes/pkg/apis/rbac",
+				"k8s.io/kubernetes/pkg/apis/rbac/v1",
+				"k8s.io/kubernetes/pkg/apis/scheduling",
+				"k8s.io/kubernetes/pkg/apis/settings",
+				"k8s.io/kubernetes/pkg/apis/settings/install",
+				"k8s.io/kubernetes/pkg/apis/settings/v1alpha1",
+				"k8s.io/kubernetes/pkg/apis/storage",
+				"k8s.io/kubernetes/pkg/capabilities",
+				"k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset",
```
Comment (luxas) [inline cmd/kubeadm/.import-restrictions]: I really hope we can get rid of this some day... Could you open an issue to track that we should get rid of the dep of the internal clientset as we are using client-go?

File: cmd/kubeadm/.import-restrictions [hunk 3]
```diff
@@ -0,0 +1,179 @@
+{
+	"Rules": [
+		{
+			"SelectorRegexp": "k8s[.]io/(api/|apimachinery/|apiextensions-apiserver/|apiserver/)",
+			"AllowedPrefixes": [
+				""
+			]
+		},
+		{
+			"SelectorRegexp": "k8s[.]io/client-go/",
```
Comment (luxas) [inline cmd/kubeadm/.import-restrictions]: This syntax is most probably right but looks strange to me. Intuitively I would have expected something like `k8s\.io\/client-go\/.*`

File: cmd/kubeadm/.import-restrictions [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+				"k8s.io/kubernetes/pkg/util/pointer",
+				"k8s.io/kubernetes/pkg/util/procfs",
+				"k8s.io/kubernetes/pkg/util/slice",
+				"k8s.io/kubernetes/pkg/util/taints",
+				"k8s.io/kubernetes/pkg/util/version",
+				"k8s.io/kubernetes/pkg/version",
+				"k8s.io/kubernetes/pkg/volume",
+				"k8s.io/kubernetes/pkg/volume/util"
+			],
+			"ForbiddenPrefixes": [
```
[the remark this answers is not in the document]
Comment (luxas) [reply cmd/kubeadm/.import-restrictions]: FWIW; isn't everything but this whitelist forbidden?
Hence it doesn't necessarily make sense to explicitely call these out here as they're automatically not permitted to be depended upon

**Review by luxas — COMMENTED** [review]
Thanks @dims! This is great
After we merge this, can you try to violate the rules in a PR so we can see it works :wink:?

File: cmd/kubeadm/.import-restrictions (hunk 3 above)
Comment (dims) [reply cmd/kubeadm/.import-restrictions]: seems to work, followed the same patter as in other .import-restrictions. Also tested by hand using

- https://play.golang.org/p/3icjV53V8it
- https://regex-golang.appspot.com/assets/html/index.html

File: cmd/kubeadm/.import-restrictions (hunk 2 above)
Comment (dims) [reply cmd/kubeadm/.import-restrictions]: yes i can

File: cmd/kubeadm/.import-restrictions (hunk 4 above)
[the remark this answers is not in the document]
Comment (dims) [reply cmd/kubeadm/.import-restrictions]: we still have `"k8s.io/kubernetes/pkg/cloudprovider"` in the AllowedPrefixes, so we need to do the forbid. (It's a prefix, not the full string)

File: cmd/kubeadm/.import-restrictions (hunk 2 above)
Comment (dims) [reply cmd/kubeadm/.import-restrictions]: https://github.com/kubernetes/kubeadm/issues/806

Comment (luxas) [reply cmd/kubeadm/.import-restrictions]: Thanks!

File: cmd/kubeadm/.import-restrictions (hunk 4 above)
[the remark this answers is not in the document]
Comment (luxas) [reply cmd/kubeadm/.import-restrictions]: Ah ok I see

File: cmd/kubeadm/.import-restrictions (hunk 2 above)
Comment (luxas) [reply cmd/kubeadm/.import-restrictions]: Thanks!

**Review by luxas — APPROVED** [review]
/lgtm

:tada:

**Pull request merged**